### DISCUSSION TITLE
Offline available desktop application

### DISCUSSION THREAD

**Discussion by mariomadproductions** [josdejong/svelte-jsoneditor#218] [Ideas]
I'd like to use this fully offline (not relying on the "install this website" feature of chrome), but I don't want to install npm. Is there a way I can do this?

**Comment by josdejong** [comment]
Are you referring to the web application https://jsoneditoronline.org? This website is offline available by default. After you've opened it once, you can just open it again when you're offline in any browser.

**Comment by mariomadproductions** [comment]
Ah yeah, the web app. I'm wondering if there's a way I can just have the html and its resources in a folder, and run it like that.

**Comment by josdejong** [comment]
That's not supported, sorry. Maybe some day I can see whether I can provide a Desktop application via [Electron](https://www.electronjs.org/) for example.

What is the reason that you need the application offline? (isn't the offline support in the browser enough?)

**Comment by mariomadproductions** [comment]
Basically I want it to be fully self-contained.

**Comment by josdejong** [comment]
Ok and can you explain why do you wan that? 

(I can understand wanting to be able to use the app without internet - but that's covered already)

**Comment by mariomadproductions** [comment]
Basically I'm not sure how persistent the chrome apps are, and I'd like to be able to make sure the code doesn't make any internet connections for privacy/security.

**Comment by josdejong** [comment]
Clear, that's a valid concern. Roughly: the web application stays offline available as long as you don't clear your browser data. It's indeed not an absolute guarantee.

If there is much demand for a desktop application I can consider it, please drop a message here then.

**Comment by mariomadproductions** [comment]
I think I'll look more into how the progressive web app/chrome "install app" options work a bit more. Maybe there's a way to copy the data it stores into a standalone folder. But thanks for the offer.

**Comment by josdejong** [comment]
👍 I'll close this issue now and turn it into a discussion. Thanks for your input!

**Reply by mariomadproductions** [reply]
Could the issue be recreated? It doesn't seem like using the chrome "install app" option provides something that is necessarily persistent, nor can be isolated from the internet.

**Reply by josdejong** [reply]
Sure, this discussion is open :)
